### DISCUSSION TITLE
LED Matrix artifacts

### DISCUSSION THREAD

**Discussion by jekader** [pixelit-project/PixelIt#339] [General]
Hi everyone, I'm working on porting PixelIt to ESP32-C3 and so far it's looking good.
Using the awesome [MicroMatrix](https://github.com/foorschtbar/MicroMatrix) and the boot logo is rendered perfectly, however later I see some artifacts:

![crop](https://github.com/pixelit-project/PixelIt/assets/2088123/338c358c-02b1-4678-9f54-f6b8e68846f3)

The settings are:
* startup brightness: 2
* auto brightness: off
* clock color: #00FF00 (green)
* clock auto fallback: 30 seconds, no animation
* LED matrix pin: D2

Has anyone stumbled across similar glitches before? Some of my theories are:
* insufficient power, however the boot logo prints without issues
* timing issues
* something else using the pin and corrupting the data stream

Any advice welcome before I get the soldering iron out.

**Comment by foorschtbar** [comment]
What you could try to figure out if it is a hardware or software issue:

- 1000uF cap between 5V and GND
- external 5V/2A power supply
- Generic ESP8266 with default PixelIt pins
- NeoPixel test sketch

**Comment by jekader** [comment]
Thanks for the advice. After some more tests looks like it is something specific to FastLED on ESP32-C3, there's a bug reported upstream for quite a while: https://github.com/FastLED/FastLED/issues/1349

When setting `MATRIX_PIN=8` to send control signals to the built-in blue LED, I still see it flickering even when a static text gets displayed and no LED refresh should take place. Might be signal interference from Bluetooth/WiFi, will keep on digging.

**Comment by Linjieqiang** [comment]
LGTM

**Reply by Linjieqiang** [reply]
Hello @jekader , how to test it? I have an ESP32-mini board, but I'd like to get the firmware and test it

**Reply by jekader** [reply]
Go ahead and clone this branch and build it with PlatformIO for ESP32-C3:
https://github.com/jekader/PixelIt/tree/esp32c3

This should work and reproduce the issue (flicker on wifi activity due to single-core delays), so change line [153 of platformio.ini](https://github.com/pixelit-project/PixelIt/compare/main...jekader:PixelIt:esp32c3#diff-4446afd728a4f34cbcddc306a9cb6be845d1a61c216076a295683bcc9c106724R153) to `fastled/FastLED` (remove version), then rebuild and reupload to confirm the issue is fixed.

**Reply by Linjieqiang** [reply]
Good, let me try
